lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -134,6 +134,28 @@ def create(path)
       @zk.create(path, ignore: :node_exists)
     end
 
+    # get with retry to reduce failure when network connectivity is flaky or ZK is overloaded
+    def get(path, opts = {})
+      retry_limit = opts[:retry_limit] || 3
+      opts.delete(:retry_limit)
+      retry_interval = opts[:retry_interval] || 5
+      opts.delete(:retry_interval)
+      retry_count = 0
+      begin
+        if retry_count > retry_limit
+          raise RuntimeError, "synapse: get ZK path failed after retry"
+        end
+        retry_count += 1
```

Comment:
this should happen before `if retry_count > retry_limit`? actually, you will end up trying `4` times in total now. Or maybe you actually mean try once first then retrying another 3 times? So to make logic simple, you could move retry check into `rescue` part. Following example is trying up to `4` times.

```
retry_count += 1
if retry_count > retry_limit
   raise RuntimeError, "synapse: get ZK path failed after retry"
else
  sleep retry_interval
  retry
end
```

Reply:
retry_limit means how many times it should retry on top of the initial try. Let me document the input to make it obvious.

> also, remember to bump version

Sure